propnet/ext/matproj.py

```diff
@@ -22,104 +22,70 @@
     'final_energy_per_atom': 'final_energy_per_atom',
     'formation_energy_per_atom': 'formation_energy_per_atom',
     'piezoelectric_tensor': 'piezo.piezoelectric_tensor',
-    'volume_unit_cell': 'volume',
-    'structure': 'structure'
+    'volume_unit_cell': 'volume'
 }
 PROPNET_FROM_MP_NAME_MAPPING = {v: k for k, v in MP_FROM_PROPNET_NAME_MAPPING.items()}
 
 # list of all available properties
-AVAILABLE_MP_PROPERTIES = list(PROPNET_FROM_MP_NAME_MAPPING.keys()) + ['task_id']
+AVAILABLE_MP_PROPERTIES = list(PROPNET_FROM_MP_NAME_MAPPING.keys()) + ['task_id', 'structure']
 PROPNET_PROPERTIES_ON_MP = list(MP_FROM_PROPNET_NAME_MAPPING.keys())
 
 
-def import_all_props(mp_ids):
-    """Given a list of materials ids, returns a list of all available properties
-    for each material. One list per material id is stored in a list of lists.
-
-    Args:
-      mp_ids: materials ids for which properties are requested.
-
-    Returns:
-      list of lists containing all possible materials properties by mp_id
-
+def import_materials(mp_ids, api_key=None):
     """
-    return import_props(mp_ids, AVAILABLE_MP_PROPERTIES)
-
-
-def import_props(mp_ids, prop_list):
-    """Given a list of materials ids and mp property names returns a list of property
-    instances for each requested material. These lists are stored in an encapsulating
-    list.
-
+    Given a list of material ids, returns a list of Material objects with all
+    available properties from the Materials Project.
     Args:
-      mp_ids: materials ids for which properties are requested.
-      prop_list: requested MP properties for materials.
-
+        mp_ids (list<str>): list of material ids whose information will be retrieved.
+        api_key (str): api key to be used to conduct the query.
     Returns:
-      list of lists containing requested materials properties by mp_id
-
+        (list<Material>): list of material objects with associated data.
     """
-    query = mpr.query(criteria={"task_id": {'$in': mp_ids}}, properties=prop_list)
-    print(query)
-    # properties of all mp-ids inputted; list of lists
-    properties = []
+    if api_key:
```

Comment:
get rid of this if statement -- just pass api_key directly as mpr = MPRester(api_key) -- if it's `None` (as default), pymatgen will check env variables or config files for the api key as normal

Reply:
done